Apromore-Core-Components/apromore-storage/src/test/java/org/apromore/test/storage/StorageFactoryUnitTest.java

```diff
@@ -46,11 +46,12 @@ public void setup()
     
     
     @Test
-    public void testFileStorageFactory() throws ObjectNotFoundException
+    public void testFileStorageFactory() throws ObjectNotFoundException, URISyntaxException
     {
 //	Given
 	ClassLoader classLoader = StorageFactoryUnitTest.class.getClassLoader();	
-	File file = new File(classLoader.getResource("baseFolder"+File.separator+"test.csv").getFile());
+	//File file = new File(classLoader.getResource("baseFolder"+File.separator+"test.csv").getFile());
+	File file = new File(classLoader.getResource("baseFolder"+File.separator+"test.csv").toURI());
```

Comment:
I don't think it's necessary to use the system-specific File.separator because this is a Java classpath rather than an actual pathname on the native filesystem.  Could you check if it still works on Windows to simply use "baseFolder/test.csv"?

Reply:
Sure it will work because no difference. But File.separator is usually recommended as a safe OS-independent although recent Java versions can do the same for the "/" character. However, the classpath on Windows and Unix still have differences.

Reply:
Actually you're right Simon. File.separator is the root cause of this issue. On Windows, File.separator will be converted to backslash "\" and then this character in the file path will be converted to "%5c", so the file path will contain "baseFolder%5ctest.csv" which was the problem of File not found. Using forward-slash "baseFolder/test.csv" and Java will convert it to a correct system-specific format, on Windows it is "baseFolder\\test.csv". Call URL.toURI() can fix the strange character "%5c" in the path but it is not the root cause.

Reply:
That looks good now.  👍